workflow/controller/controller.go

```diff
@@ -217,6 +225,24 @@ func (wfc *WorkflowController) Run(ctx context.Context, wfWorkers, workflowTTLWo
 	}
 	logCtx := log.WithField("id", nodeID)
 
+	// Create leader election health checker
+	leaderElectionChecker := leaderelection.NewLeaderHealthzAdaptor(leaderHealthzAdaptorTimeout)
+	var checks []healthz.HealthChecker = nil
+	checks = append(checks, leaderElectionChecker)
+	mux := http.NewServeMux()
+	healthz.InstallPathHandler(mux, "/healthz", checks...)
+	healthCheckServer := &http.Server{
+		Addr:    fmt.Sprintf(":%d", healthCheckPort),
+		Handler: mux,
+	}
+	go func() {
```

Comment:
Do we need this for standby pod too? otherwise can you move into `leaderelection.RunOrDie`?

Reply:
Sure. Updated and rebased.

Reply:
Actually yes we need this for standby pod too so it's better to keep this outside of `RunOrDie`.

Reply:
Sure. My only concern is here if somebody does not require the health check, it will unnecessarily listening.  Can we have a command line flag/config item to enable leaderelection healthcheck and port? So, if anybody wants, they can enable it

Reply:
That's a good point. Just added a flag to toggle this.